Dockerfile

```diff
@@ -0,0 +1,14 @@
+ARG SPARK_VERSION=2.4.4
+FROM fandango.azurecr.io/opti_spark:${SPARK_VERSION}
+
+RUN mkdir -p /usr/share/man/man1
+
+RUN apt-get update
+RUN yes | apt-get install -y build-essential maven
+
+COPY Makefile ./
+COPY VERSION ./
+COPY src ./src
+COPY pom.xml ./
+RUN make build
+ENTRYPOINT ["/var/lib/spark/bin/spark-submit"]
```

Comment:
I think the entrypoint should hide details to the caller.
Right now if I want to use this container I need to know where is the built jar located within the container.
a caller should not be aware of that, a caller concern should only be giving the input and output locations. 

We will need to fix this later anyway to provide a bootstrap script for AZ and to handle upload/downloads from cloud paths so I am happy to tick as it is now

Reply:
what he said